src/test/gtest_main.cpp

```diff
@@ -1,19 +1,58 @@
-#include <sys/stat.h>
-#include <sys/types.h>
+#if __has_include(<filesystem>)
+#include <filesystem>
+#else
+#include <experimental/filesystem>
+#endif
 
 #include <gtest/gtest.h>
+#include "base_test.hpp"
 
 #include "utils/assert.hpp"
 #include "utils/performance_warning.hpp"
 
+#if __has_include(<filesystem>)
+namespace filesystem = std::filesystem;
+#else
+namespace filesystem = std::experimental::filesystem;
+#endif
+
+void create_test_data_directory(std::optional<std::string>& prefix) {
+  if (prefix)
+    opossum::TEST_DATA_PATH = "./" + *prefix + "/test_data/";
+  else
+    opossum::TEST_DATA_PATH = "./test_data/";
+
+  if (filesystem::exists(opossum::TEST_DATA_PATH)) {
+    filesystem::remove_all(opossum::TEST_DATA_PATH);
+  }
+  filesystem::create_directories(opossum::TEST_DATA_PATH);
+}
+
+void remove_test_data_directory(std::optional<std::string>& prefix) {
+  if (prefix)
+    filesystem::remove_all("./" + *prefix);
+  else
+    filesystem::remove_all(opossum::TEST_DATA_PATH);
```

Comment:
Can't we use `TEST_DATA_PATH` in any case?

Reply:
No, in case a prefix is given, only the folder `test_data` in `./prefix/` would be deleted, but we want `./prefix` to be deleted. However, if somebody specifies a prefix consisting of multiple folders, we will run into the same problem

Reply:
That makes the behavior even more surprising. :/ In one case, we delete the `test_data` folder, in the other case, we delete the folder that contains `test_data`.

Reply:
The naming is problematic, but we do want to delete all folders which have been created during testing?!